### PR TITLE
tests: Run through shellcheck

### PR DESCRIPTION
**Pull request by neheb**
Various fixes.

Grouped several writes. Maybe that speeds up something...

### CODE REVIEW

**Comment by neheb** [comment]
Several bugs were also fixed (TOTTALY_SILENT misspelling comes to mind).

File: test/common.sh.in [hunk 1]
```diff
@@ -51,7 +52,7 @@ top_srcdir=@top_srcdir@
 top_builddir=@top_builddir@
 
 # Set `is_win` variable which is used in other scripts that source this one.
-if test $(env | grep -ic '^comspec=') != 0 ; then
+if [ "$(env | grep -ic '^comspec=')" != 0 ]; then
```
Comment (erikd) [inline test/common.sh.in:55]: i disagree with replacing `test` with `[` and `]`.

**Comment by ktmf01** [comment]
@neheb Would you mind taking a look at this once more? There have been quite a few additions to the test suite, so maybe you can redo this PR with the newest version of shellcheck with the current git master?

**Comment by neheb** [comment]
Done.

**Comment by neheb** [comment]
hmmm maybe I should split this up into particular fixes.

**Comment by ktmf01** [comment]
@neheb Can you let me know when this is finished? Or is it already? I can't really tell

**Comment by neheb** [comment]
Might as well merge. The rest of the "issues" shellcheck complains about deal with quoting. That's something I'd rather not touch right now.

**Comment by ktmf01** [comment]
Thanks!

**Pull request merged**